src/ray/gcs/gcs_server/gcs_function_manager.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "absl/container/flat_hash_map.h"
+#include "ray/gcs/gcs_server/gcs_kv_manager.h"
+
+namespace ray {
+namespace gcs {
+
+class GcsFunctionManager {
```

Comment:
Add a comment about purpose of the class, and maybe example usage? Maybe move the comment above `job_counter_` here.